docs/specs/6-auth-via-aad.md

```diff
@@ -0,0 +1,168 @@
+var token = await credential.GetTokenAsync(context);
+```
+
+#### 2. Managed identity (user-assigned)
+
+> Endpoint=(endpoint);AuthType=aad;ClientId=(managed identity client id);Version=1.0;
+
+##### Sample code (C#)
+
+```C#
+var credential = new ManagedIdentityCredential(clientId);
+var context = new TokenRequestContext(new string[] { "https://signalr.azure.com/.default" });
+var token = await credential.GetTokenAsync(context);
+```
+
+#### 3. Azure application 
+
+> Endpoint=(endpoint);AuthType=aad;ClientId=(application client id);TenantId=(application tenant id);ClientSecret=(application client secret);Version=1.0;
+
+##### Sample code (C#)
+
+```C#
+var credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
+var context = new TokenRequestContext(new string[] { "https://signalr.azure.com/.default" });
+var token = await credential.GetTokenAsync(context);
+```
+
+## Websocket connection
+
+For short, instead of generating an `AccessToken` with `AccessKey` while processing negotiation requests,
+the server could request an `AccessToken (ClientToken)` from the Web PubSub service with an `AadToken`.
+
+### Workflow
+
+![workflow](../images/aad-websocket-connection-workflow.png)
+
+1. Client sends a negotiate request to App to get a `AccessToken`.
+
+2. **App calls [MSAL library](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-overview), with credentials in the connection string.**
+
+3. **App receives the `AadToken` from Azure authentication endpoint.**
+
+    _The ADD Token contains identity info of the service principal such as objectId, tenantId, groups._
+
+4. **App calls GET `AccessToken` API, align with the `AadToken`.**
+
+5. Service sends a request to Azure resource manager endpoint _(rbac)_.
+
+6. Service receive permitted actions.
+
+7. Service check if the service principal had the _clientToken/action_ permission. If so, get or generate a temporary `AccessKey`, then sign a `AccessToken` with the `AccessKey`.
+
+    _The key id of AccessKey will be associated with the objectId, which means we have the ability to know from which service principal the AccessToken has been signed._
+
+8. **App receives the `AccessToken`.**
+
+9. Client receives the `AccessToken`.
+
+10. Client creates WebSocket connection with the `AccessToken`.
+
+11. Service check if the `AccessToken` was signed by known `AccessKeys` and check if the service principal had the _clientConection/write_ permission.
+
+12. Client receive 101 or 401 response.
+
+### REST API design (ClientToken)
+
+#### GET /auth/clientToken
+
+##### URL
+
+`https://<service endpoint>/api/v1/auth/clientToken`
+
+##### Request
+
+| Name | Type | Required | Sample | Description |
+|---|---|---|---|---|
+| `hub` | string | \* | chat | The hub name |
+| `userId` | string | | foo | The userId for the client |
+| `roles` | string | | webpubsub.joinLeaveGroup,webpubsub.sendToGroup | The roles that the client will have, separated by comma. |

```

Comment:
follow the same pattern as https://github.com/Azure/azure-rest-api-specs/blob/master/specification/webpubsub/data-plane/WebPubSub/preview/2021-05-01-preview/webpubsub.json#L60